### PR TITLE
improved sql comment handling

### PR DESCRIPTION
**Pull request by igorwwwwwwwwwwwwwwwwwwww**
This is an improved version of https://github.com/basecamp/marginalia/pull/56.

### CODE REVIEW

File: lib/marginalia/comment.rb [hunk 1]
```diff
@@ -26,9 +26,18 @@ def self.construct_comment
         end
       end
       ret.chop!
+      ret = self.escape_sql_comment(ret)
       ret
     end
 
+    def self.escape_sql_comment(str)
+      str = str.dup
```
Comment (arthurnn) [inline lib/marginalia/comment.rb]: Can't we only `dup` if the `str` includes the patterns bellow? I want to save the extra allocation if not necessary.

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
I've made the dup conditional. In fact I'm not sure if it is even needed. Since the `gsub` should not modify the original. So we may be able to drop the `dup` completely.

File: lib/marginalia/comment.rb [hunk 2]
```diff
@@ -26,9 +26,20 @@ def self.construct_comment
...
       end
       ret.chop!
+      ret = self.escape_sql_comment(ret)
       ret
     end
 
+    def self.escape_sql_comment(str)
+      if str.include?('/*') || str.include?('*/')
+        str = str.dup
+      end
```
Comment (arthurnn) [inline lib/marginalia/comment.rb]: You are right, we don't need the `dup`. gsub should already create a new string for us.

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
Removed the `dup`.

**Comment by arthurnn** [comment]
Thanks. i am cleaning up test and releasing a new gem version

**Pull request merged**